sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,299 @@
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/crs"
+	"github.com/stackrox/rox/pkg/env"
+	grpcUtil "github.com/stackrox/rox/pkg/grpc/util"
+	"github.com/stackrox/rox/pkg/k8sutil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/pods"
+	protoconv "github.com/stackrox/rox/pkg/protoconv/certs"
+	"github.com/stackrox/rox/pkg/protoutils"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/version"
+	"github.com/stackrox/rox/sensor/common/centralclient"
+	sensorCommon "github.com/stackrox/rox/sensor/common/sensor"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/securedcluster"
+	"github.com/stackrox/rox/sensor/kubernetes/helm"
+	"github.com/stackrox/rox/sensor/kubernetes/sensor"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	log                                  = logging.LoggerForModule()
+	fetchSensorDeploymentOwnerRefBackoff = wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   3,
+		Jitter:   0.1,
+		Steps:    10,
+		Cap:      6 * time.Minute,
+	}
+)
+
+// EnsureServiceCertificatesPresent initiates the CRS-based cluster registration flow to retrieve the
+// service certificates in case no services certificates are found and instead a CRS is present.
+func EnsureServiceCertificatesPresent() error {
+	log.Infof("Ensuring certificates for Secured Cluster services are present.")
+	clientconn.SetUserAgent(fmt.Sprintf("%s Cluster Registration Helper", clientconn.Sensor))
+
+	// Check if legacy sensor service certificate, e.g. created via init-bundle, exists.
+	if legacySensorServiceCert := os.Getenv(crs.LegacySensorServiceCertEnvName); legacySensorServiceCert != "" {
+		log.Infof("Legacy sensor service certificate available, skipping cluster registration using Cluster Registration Secret.")
+		return nil
+	}
+
+	// Check if modern sensor service certificate, e.g. created by CRS-based cluster registration or
+	// refreshed by periodic cert rotation, exists.
+	if sensorServiceCert := os.Getenv(crs.SensorServiceCertEnvName); sensorServiceCert != "" {
+		log.Infof("Sensor service certificate available, skipping CRS-based cluster registration.")
+		return nil
+	}
+
+	log.Infof("No sensor service certificate found, initiating cluster registration using Cluster Registration Secret.")
+	if err := registerCluster(); err != nil {
+		return errors.Wrap(err, "registering secured cluster")
+	}
+	log.Info("Cluster registration complete.")
+
+	return nil
+}
+
+// registerCluster implements the CRS-based registration flow for secured clusters.
+// It
+//   - retrieves the CRS secret
+//   - unpacks the contained mTLS certificate+key pair for the REGISTRANT_SERVICE identity from the
+//     CRS and stores it in /run/secrets/stackrox.io/certs (which must be a memory-backed
```

Comment:
the `memory-backed` part is optional, right?

Reply:
Well, it *shall* be memory-backed to prevent writing the key to disk.

Reply:
Oh, if that's the intention of _memory_-backing, then it makes sense to mention this in a comment here or in the manifest!

Reply:
Sorry, not sure what you mean. Is this resolved then or would you like some addition do the comment or is it ok like this?

Reply:
What I mean is that there are two concerns here: 
1.  that `/run/secret/stackrox.io/certs` is a writable location - this is different from how this directory is set up in regular sensor, therefore it is worthy of mentioning,
2. that making this location memory backed protects from persisting the secrets to disk. It might be that a comment about this concern should be in the place where the mount is set up that is the manifest template, rather than here.

Anyway, just a nit, up to you.